v3/ldap.go

```diff
@@ -82,6 +83,13 @@ var BeheraPasswordPolicyErrorMap = map[int8]string{
 	BeheraPasswordInHistory:           "New password is in list of old passwords",
 }
 
+var logger = log.Default()
+
+// Logger allows clients to override the default logger
+func Logger(l *log.Logger) {
+	logger = l
+}
```

Comment:
Out of curiosity; why a private variable and helper function vs a public variable like the stdlib does?

Reply:
I can't find the example in the stdlib I was thinking of... ignore this :)

Reply:
Great question. Encapsulation. Making the logger private to the package, it cannot be directly used (or abused) outside of the library.